src/test/kotlin/uk/gov/justice/digital/hmpps/adjustments/api/legacy/controller/LegacyControllerIntTest.kt

```diff
@@ -70,10 +65,39 @@ class LegacyControllerIntTest : IntegrationTestBase() {
 
     val legacyData = objectMapper.convertValue(adjustment.legacyData, LegacyData::class.java)
     assertThat(legacyData).isEqualTo(LegacyData(bookingId = 1, sentenceSequence = 1, comment = "Created", type = LegacyAdjustmentType.UR, active = true))
+
+    assertThat(getNumberOfMessagesCurrentlyOnQueue()).isEqualTo(1)
+    val latestMessage = getLatestMessage()!!.messages[0].body
+    assertThat(latestMessage).contains(adjustment.id.toString())
+    assertThat(latestMessage).contains(EventType.ADJUSTMENT_CREATED.value)
+    assertThat(latestMessage).contains(AdjustmentSource.NOMIS.name)
+  }
+
+  @Test
+  fun migration() {
+    cleanQueue()
+    val result = webTestClient
+      .post()
+      .uri("/legacy/adjustments/migration")
+      .headers(
+        setAuthorisation()
+      )
+      .header("Content-Type", LegacyController.LEGACY_CONTENT_TYPE)
+      .bodyValue(CREATED_ADJUSTMENT)
+      .exchange()
+      .expectStatus().isCreated
+      .returnResult(LegacyAdjustmentCreatedResponse::class.java)
+      .responseBody.blockFirst()!!
+
+    val adjustment = adjustmentRepository.findById(result.adjustmentId).get()
+    assertThat(adjustment).isNotNull
+
+    assertThat(getNumberOfMessagesCurrentlyOnQueue()).isEqualTo(0)
```

Comment:
So am I right in thnking that the POST drops a message on a queue which is processed to create `adjustmentId`? If so I think this test might prove to be flaky... we normally use awaitility to make sure that the message has been processed before checking the results of the message. Something like [this](https://github.com/ministryofjustice/hmpps-spring-boot-sqs/blob/main/test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/HmppsQueueResourceTest.kt#L81) which should check that queue attribitutes `ApproximateNumberOfMessages` and `ApproximateNumberOfMessagesNotVisible` (in flight) are both zero - hence the we know that the message has been consumed.

Reply:
The POST creates an adjustment, then once the transaction has completed a message is placed on the queue. This all happens before the POST request ends. Would the message be on the queue immediately when this test runs then? I've not seen any intermittent failures yet, but I can implement awaitility if needed.